File: julia/formatter.jl
"""
arraydata = formatter(filename::String, nx::Int, ny::Int; HeaderLines::Int=0)
### Arguments
- filename : file path and file name
- nx, ny   : number of x- and y- grids
### Keyword argument
- HeaderLines : number of header lines. Default is zero (no header).
### Return value
- arraydata: mesh data formatted from fixed length array
"""
function formatter(filename::String, nx::Int, ny::Int; HeaderLines::Int=0)
    txt = readlines(filename)
    arraydata = [parse(Float64, txt[i+HeaderLines][8(j-1)+1:8j]) for i=1:length(txt), j=1:10]
    arraydata = permutedims(arraydata, [2,1])
    arraydata = reshape(vec(arraydata), (nx, ny))
    arraydata = reverse(permutedims(arraydata, [2,1]), dims=1)
    return arraydata
end

File: matlab/formatter.m
function arraydata = formatter(varargin)
% arraydata = formatter(filename, nx, ny, HeaderLines)
% input args
%   filename    : file path
%   nx, ny      : number of x- and y- grids
%   HeaderLines : number of header lines
%  
narginchk(3,5)

filename = varargin{1};
nx = varargin{2};
ny = varargin{3};

if nargin < 4
    HL = 0;
else
    HL = varargin{4};
end
if nargin < 5
    fmt = '%8.2f';
else
    fmt = varargin{5};
end

fid = fopen(filename,'r');
org = textscan(fid,fmt,'HeaderLines',HL);
fclose(fid);
org = cell2mat(org);

arraydata = flipud(permute(reshape(org,[nx,ny]),[2,1]));

end
